backend/src/main/java/planit/people/preparation/APIs/API_Calendar.java

```diff
@@ -73,17 +78,70 @@ public ResponseEntity<List<DTO_PresetDetail>> getAllPresetFromPlanItUserId(@Path
     }
 
     /**
-     * NOT COMPLETE
+     * Update an already existing EventPreset
      *
      * @param dtoPresetDetail the body of the updated preset detail.
-     * @return Status OK
+     * @param planItUserId    the id of the PlanIt User who requested the update
+     * @return the updated preset detail
+     * @see Service_Calendar#upsertPresetDetail(Long, DTO_PresetDetail, Boolean)
      */
-    @PatchMapping
-            (path = "presets", consumes = {MediaType.APPLICATION_JSON_VALUE})
-    public ResponseEntity<Boolean> updateNewPreset(@RequestBody DTO_PresetDetail dtoPresetDetail) {
+    @PatchMapping(path = "presets/{planit-user-id}", consumes = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<DTO_PresetDetail> updatePreset(@RequestBody DTO_PresetDetail dtoPresetDetail, @PathVariable("planit-user-id") Long planItUserId) {
         try {
             System.out.println("request: " + dtoPresetDetail);
-            serviceCalendar.upsertPresetDetail(null, dtoPresetDetail, true);
+            return new ResponseEntity<>(serviceCalendar.upsertPresetDetail(planItUserId, dtoPresetDetail, true), HttpStatus.OK);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * Delete an EventPreset instance for a given PlanItUser
+     *
+     * @param planItUserId the id of the PlanIt user whose instance should be removed.
+     * @param presetId     the id of the EventPreset which should be removed.
+     * @return HttpStatus.OK if deleted else HttpStatus.BAD_REQUEST
+     */
+    @DeleteMapping(path = "presets/{planit-user-id}/{preset-id}")
+    public ResponseEntity<Boolean> deletePreset(@PathVariable("planit-user-id") Long planItUserId, @PathVariable("preset-id") Long presetId) {
+        try {
+            serviceCalendar.deletePreset(planItUserId, presetId);
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * share an EventPreset with another PlanIt User
+     *
+     * @param sharePreset contains the sharing information
+     * @return HttpStatus.OK if invite email is sent else HttpStatus.BAD_REQUEST
+     */
+    @PostMapping(path = "presets/share", consumes = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Maybe let's make it `presets/{preset-id}/share`?

Reply:
I would leave the preset id in the body of the request. in order to have all parameters in one place